azurerm/internal/services/apimanagement/api_management_custom_domain_resource.go

```diff
@@ -305,7 +307,7 @@ func expandApiManagementCustomDomains(input *schema.ResourceData) *[]apimanageme
 	return &results
 }
 
-func flattenApiManagementHostnameConfiguration(input *[]apimanagement.HostnameConfiguration, d *schema.ResourceData) []interface{} {
+func flattenApiManagementHostnameConfiguration(input *[]apimanagement.HostnameConfiguration, d *schema.ResourceData, name, apimHostNameSuffix string) []interface{} {
```

Comment:
I think it would be cleaner to pass in the full hostname to be ignored, and determine that hostname before calling `flattenApiManagementHostnameConfiguration()`. If it accepts a slice that effectively makes it optional, and allows for potential future additions relatively easily.

```suggestion
func flattenApiManagementHostnameConfiguration(input *[]apimanagement.HostnameConfiguration, ignoreHostname []string) []interface{} {
```

Reply:
@manicminer thanks for the idea. 

I could move the whole filtering logic on layer up into the `apiManagementCustomDomainRead` function then it wouldn't be required to pass the `ignoreHostname[]` into the flatten  function at all.

One thing to consider: Right now it is exactly the same implementation as in the `api_management_resource` (see: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/apimanagement/api_management_resource.go#L1019 ). My original idea was to keep the logic and code similar to that.

What do you think?

Reply:
@patst I'd forgotten that the parent resource had the same logic - in that case let's leave your original implementation as-is for now and disregard my suggestion :)